src/require-strict-assert.spec.ts

```diff
@@ -0,0 +1,107 @@
+    valid: [
+      {
+        code: `import { strict as assert } from 'node:assert';`,
+      },
+      {
+        code: `import { strict as assert } from 'node:assert';
+               import otherModule from 'other-module';`,
+      },
+    ],
+    invalid: [
+      {
+        code: `import * as assert from 'node:assert';`,
+        errors: [
+          {
+            message: 'Require the strict version of node:assert.',
+          },
+        ],
+        output: `import { strict as assert } from 'node:assert';`,
+      },
+      {
+        code: `import assert from 'node:assert';`,
+        errors: [
+          {
+            message: 'Require the strict version of node:assert.',
+          },
+        ],
+        output: `import { strict as assert } from 'node:assert';`,
+      },
+      {
+        code: `import * as assert from 'node:assert';
+               import otherModule1 from 'other-module1';
+               import otherModule2 from 'other-module2';`,
+        errors: [
+          {
+            message: 'Require the strict version of node:assert.',
+          },
+        ],
+        output: `import { strict as assert } from 'node:assert';
+               import otherModule1 from 'other-module1';
+               import otherModule2 from 'other-module2';`,
+      },
+      {
+        code: `import assert from 'node:assert';
+               assert.strictEqual(value1, value2);`,
+        errors: [
+          {
+            message: 'Require the strict version of node:assert.',
+          },
+          {
+            message: 'Use non-strict counterpart for assert function.',
+          },
+        ],
+        output: `import { strict as assert } from 'node:assert';
+               assert.equal(value1, value2);`,
+      },
+      {
+        code: `import { strict as assert } from 'node:assert';
+               assert.strictEqual(value1, value2);`,
+        errors: [
+          {
+            message: 'Use non-strict counterpart for assert function.',
+          },
+        ],
+        output: `import { strict as assert } from 'node:assert';
+               assert.equal(value1, value2);`,
+      },
+      {
+        code: `import { strict as assert } from 'node:assert';
+               assert.deepStrictEqual(obj1, obj2);`,
+        errors: [
+          {
+            message: 'Use non-strict counterpart for assert function.',
+          },
+        ],
+        output: `import { strict as assert } from 'node:assert';
+               assert.deepEqual(obj1, obj2);`,
+      },
+      {
+        code: `import { strict as assert } from 'node:assert';
+               const val1 = 'val2';
+               assert.strict(val1, 'val2');`,
+        errors: [
+          {
+            message: 'Use non-strict counterpart for assert function.',
+          },
+        ],
+        output: `import { strict as assert } from 'node:assert';
+               const val1 = 'val2';
+               assert.equal(val1, 'val2');`,
+      },
+    ],
```

Comment:
We should also account for `import assert from 'node:assert/strict`.  Maybe something with a fix to our standard `import { strict as assert } etc` style